src/Model/OrderLabel.php

```diff
@@ -399,9 +400,13 @@ static function () use ($orderId) {
                     ], OrderLogger::WARNING);
                 }
 
-                if (! ($result[0]['email'] ?? false)) {
+                $emptyFields = Utils::getKeysWithoutValue(reset($result), Constant::REQUIRED_LABEL_KEYS);
+                if ($emptyFields) {
                     throw new MissingFieldException(
-                        sprintf('Customer not found for order %s', $orderId)
+                        sprintf(
+                            'Customer information in the following fields are missing: %s',
```

Comment:
Waarom niet gewoon zo?
```suggestion
                            'The following fields are missing but required: %s',
```